doc/flux_to_redux.md

```diff
@@ -0,0 +1,641 @@
+
+```dart
+import 'package:w_flux/w_flux.dart' as flux;
+
+// A flux store responsible for displaying a random background color.
+class RandomColorStore extends flux.Store {
+    /// The collection of actions this store will be subscribing to.
+    RandomColorActions _actions;
+
+    /// Public data
+    String _backgroundColor = 'gray';
+    String get backgroundColor => _backgroundColor;
+
+    RandomColorStore(this._actions) {
+    // Subscribe to an action. When the action fires:
+    // 1. call _changeBackgroundColor
+    // 2. trigger an update
+        triggerOnActionV2(_actions.changeBackgroundColor, _changeBackgroundColor);
+    }
+
+    _changeBackgroundColor(String _) {
+        // Update the state, which will be reflected in the public getter
+        _backgroundColor = '#' + (Random().nextDouble() * 16777215).floor().toRadixString(16);
+    }
+}
+```
+
+```dart
+import 'package:w_flux/w_flux.dart' as flux;
+
+/// A collection of actions.
+class RandomColorActions {
+  /// An action instance that can be used to dispatch events to subscribers.
+  final changeBackgroundColor = flux.Action<String>();
+}
+```
+
+```dart
+// This same actions object will also be passed into components,
+// so that the `Action` instances used to dispatch events are the same ones
+// that have been listened to by the store.
+final randomColorActions = RandomColorActions();
+final randomColorStore = RandomColorStore(randomColorActions);
+```
+
+<img height=400 src="./FluxArchitecture.png" alt='Flux Architecture'>
+
+### The Redux Version
+For our example, here are our different items of interest:
+- __The State Class__, `ReduxState`
+- __Actions__, a single action class called `UpdateBackgroundColorAction`
+- __The Reducer__, `randomColorReducer`
+- __The Store Object__, `randomColorStore`
+- __The UI:__ All of our UI componentry is wrapped in a `ReduxProvider`. The provider takes in a `store` prop, to which we pass `randomColorStore`. The only component is a `BigBlock` connected component. Updates are triggered by calling `props.dispatch(UpdateBackgroundColorAction())`, and since it's a connected component (using [`mapStateToProps`](./over_react_redux_documentation.md#mapstatetoprops)) state is accessed via normal props usage.
+
+```dart
+// A regular Dart class that represents the application data model
+class RandomColorState {
+    // The state values the app needs
+    String backgroundColor;
+
+    // A constructor used to establish the default state
+    RandomColorState.defaultState() : this.backgroundColor = 'gray';
+
+    // A convenience constructor used by the reducer to return new instances of the class
+    RandomColorState.update(RandomColorState oldState, {String backgroundColor})
+        : this.backgroundColor = backgroundColor ?? oldState.backgroundColor;
```

Comment:
why isn't this just something like:
```
RandomColorState.withColor(String backgroundColor) : this.backgroundColor = backgroundColor;
```

I understand that if the internal logic somehow depended on the oldState, you'd want to include that, but this doesn't use it.

Reply:
Good point. I was leaning on using the standard way of doing things even when stuff was verbose / unused so readers would consistently see the same patterns, but I believe you're correct that it makes more sense to have the example be practical